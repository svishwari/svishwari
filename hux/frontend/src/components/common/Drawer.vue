<template>
  <v-navigation-drawer
    v-model="localDrawer"
    :right="toRight"
    :style="transition"
    :width="drawerWidth"
    app
    floating
    hide-overlay
    temporary
  >
    <v-toolbar width="100%" class="box-shadow-25">
      <v-toolbar-title class="px-6">
        <slot name="header-left"></slot>
        <slot name="header-right"></slot>
      </v-toolbar-title>
      <template v-if="expandable">
        <v-icon
          color="primary"
          @click="onExpandIconClick"
          class="cursor-pointer px-6 ml-auto"
        >
          {{ expanded ? "mdi-arrow-collapse" : "mdi-arrow-expand" }}
        </v-icon>
      </template>
    </v-toolbar>
    <v-progress-linear :active="loading" :indeterminate="loading" />

    <div class="drawer-content pa-2">
      <slot></slot>
    </div>

    <v-footer
      class="d-flex justify-space-between align-center px-6 py-5"
      absolute
      padless
      height="80"
      color="white"
      elevation="5"
    >
      <slot name="footer-left"></slot>
      <slot name="footer-right"></slot>
    </v-footer>
  </v-navigation-drawer>
</template>

<script>
export default {
  name: "Drawer",

  data() {
    return {
      localDrawer: this.value,
      expanded: false,
    }
  },

  props: {
    toRight: {
      type: Boolean,
      required: false,
      default: true,
    },

    value: {
      type: Boolean,
      required: true,
      default: false,
    },

    width: {
      type: Number,
      required: false,
      default: 600,
    },

    expandedWidth: {
      type: Number,
      required: false,
      default: 900,
    },

    expandable: {
      type: Boolean,
      required: false,
      default: false,
    },

    disableTransition: {
      type: Boolean,
      required: false,
      default: false,
    },

    loading: {
      type: Boolean,
      required: false,
      default: false,
    },
  },

  computed: {
    transition() {
      return {
        transitionDuration: this.disableTransition ? "0s" : "0.5s",
      }
    },
    drawerWidth() {
      return this.expanded ? this.expandedWidth : this.width
    },
  },

  watch: {
    value: function () {
      this.localDrawer = this.value
    },

    localDrawer: function () {
      this.$emit("input", this.localDrawer)
      if (!this.localDrawer) {
        this.$emit("onClose")
        this.reset()
      }
    },
  },

  methods: {
    onExpandIconClick: function () {
      this.expanded = !this.expanded
      this.$emit("iconToggle", this.expanded)
    },

    reset() {
      this.expanded = false
    },
  },
}
</script>

<style lang="scss" scoped>
::v-deep .v-navigation-drawer__content {
  overflow-y: hidden;
}
.drawer-content {
  height: calc(100% - 130px);
  overflow-y: auto;
}
::v-deep .v-icon.v-icon::after {
  content: none;
}
</style>
