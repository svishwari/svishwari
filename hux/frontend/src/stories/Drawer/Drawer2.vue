<template>
  <v-navigation-drawer
    v-model="localDrawer"
    right
    :style="transition"
    :width="drawerWidth"
    app
    floating
    hide-overlay
    temporary
  >
    <v-toolbar
      width="100%"
      class="drawer-header no-shadow border-bottom pl-2 pr-2"
      :height="headerHeight"
    >
      <v-toolbar-title class="title-wrap">
        <icon v-if="iconType" :type="iconType" size="38" color="primary" />
        <span class="text-h2">{{ title }}</span>
        <v-spacer></v-spacer>
      </v-toolbar-title>
      <v-spacer></v-spacer>
      <slot name="header-right"></slot>
    </v-toolbar>

    <slot name="loading">
      <v-progress-linear :active="loading" :indeterminate="loading" />
    </slot>

    <div
      class="drawer-content"
    >
      <slot></slot>
    </div>

    <v-footer
      class="drawer-footer d-flex justify-space-between align-center px-6 py-4"
      absolute
      color="white"
      elevation="5"
    >
      <div>
        <hux-button
          size="large"
          tile
          variant="secondary"
          class="btn-border box-shadow-none"
          @click="localDrawer = !localDrawer"
        >
          {{ secondaryButtonText }}
        </hux-button>
        <span v-if="footerTextField && primaryButtonText" class="ml-2">{{ footerTextField }}</span>
      </div>
      <div>
        <span v-if="footerTextField && !primaryButtonText">{{ footerTextField }}</span>
        <hux-button
          v-if="primaryButtonText"
          tile
          color="primary"
        >
          {{ primaryButtonText }}
        </hux-button>
      </div>
    </v-footer>
  </v-navigation-drawer>
</template>

<script>
import icon from "../icons/Icon2.vue"
import huxButton from "../huxButton/huxButton2.vue"

export default {
  name: "Drawer",
  components: { icon, huxButton },
  props: {
    value: {
      type: Boolean,
      required: true,
      default: false,
    },

    width: {
      type: Number,
      required: false,
      default: 600,
    },

    expandedWidth: {
      type: Number,
      required: false,
      default: 900,
    },

    expandable: {
      type: Boolean,
      required: false,
      default: false,
    },

    disableTransition: {
      type: Boolean,
      required: false,
      default: false,
    },

    loading: {
      type: Boolean,
      required: false,
      default: false,
    },
    headerHeight: {
      type: String,
      required: false,
      default: "72",
    },
    title: {
      type: String,
      required: true,
    },
    iconType: {
      type: String,
      required: false,
    },
    secondaryButtonText: {
      type: String,
      required: false,
      default: "Close",
    },
    primaryButtonText: {
      type: String,
      required: false,
    },
    footerTextField: {
      type: String, 
      required: false,
    },
  },

  data() {
    return {
      localDrawer: this.value,
      expanded: false,
    }
  },

  computed: {
    transition() {
      return {
        transitionDuration: this.disableTransition ? "0s" : "0.5s",
      }
    },

    drawerWidth() {
      return this.expanded ? this.expandedWidth : this.width
    },

    hasFooterSlots() {
      return !!(this.$slots["footer-left"] || this.$slots["footer-right"])
    },
  },

  watch: {
    value: function () {
      this.localDrawer = this.value
    },

    localDrawer: function () {
      this.$emit("input", this.localDrawer)
      if (!this.localDrawer) {
        this.$emit("onClose")
        this.reset()
      }
    },
  },

  methods: {
    onExpandIconClick: function () {
      this.expanded = !this.expanded
      this.$emit("iconToggle", this.expanded)
    },

    reset() {
      this.expanded = false
    },
  },
}
</script>

<style lang="scss" scoped>
$drawer-header-height: 72px;
$drawer-footer-height: 72px;
$drawer-data-table-padding: 9px 25px;

::v-deep .v-navigation-drawer__content {
  overflow-y: hidden;
}
.drawer-header {
  ::v-deep > .v-toolbar__content {
    height: $drawer-header-height !important;
    border-bottom: 1px solid var(--v-black-lighten3);
  }
}
.drawer-footer {
  height: $drawer-footer-height;
  box-shadow: none !important;
  border-top: 1px solid var(--v-black-lighten3) !important;
}
.drawer-content {
  height: calc(100% - #{$drawer-header-height + $drawer-footer-height});
  overflow-y: auto;
  border-bottom: 1px solid var(--v-black-lighten3) !important;
}
.drawer-content-without-footer {
  height: calc(100% - #{$drawer-header-height});
}
::v-deep .v-icon.v-icon::after {
  content: none;
}

::v-deep .hux-data-table {
  .v-data-table {
    > .v-data-table__wrapper {
      > table {
        > thead {
          > tr {
            > th {
              background: var(--v-primary-lighten2) !important;
            }
          }
        }

        > thead > tr > th,
        > tbody > tr > td {
          padding: $drawer-data-table-padding;
        }
      }
    }
  }
}
.title-wrap {
  display: flex;
  align-items: center;
  justify-content: space-between;
}
</style>
