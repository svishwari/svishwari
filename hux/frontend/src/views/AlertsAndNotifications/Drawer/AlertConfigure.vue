<template>
  <drawer
    v-model="localDrawer"
    :content-padding="'pa-0'"
    :content-header-padding="'px-3'"
  >
    <template #header-left>
      <div class="d-flex align-center">
        <icon
          type="setting-gear-border"
          :size="32"
          color="black"
          class="d-block mr-2 black-border"
        />
        <h3 class="text-h2 ml-1 black--text text--darken-4">
          Configure alerts
        </h3>
      </div>
    </template>
    <template #default>
      <div class="px-6 py-4">
        <div class="text-body-1 pb-4">
          Configure if you wish to recieve Hux notifications and alerts via
          email.
        </div>
        <div class="d-flex full-alert text-body-1 pb-4">
          Do you wish to receive alerts?
          <hux-switch
            v-model="showAlerts"
            false-color="var(--v-black-lighten4)"
            width="57px"
            :switch-labels="switchLabelFullAlerts"
            class="w-53"
            @change="toggleMainSwitch($event)"
          />
        </div>
        <div v-if="showAlerts">
          <div class="pb-4 h-77">
            <tooltip max-width="300px" position-top>
              <template #label-content>
                Email
                <icon
                  type="info"
                  :size="8"
                  class="mb-1"
                  color="primary"
                  variant="base"
                />
              </template>
              <template #hover-content>
                <span
                  v-bind.prop="
                    formatInnerHTML(
                      'Email address is pre-populated from your profile and canâ€™t be modified.'
                    )
                  "
                />
              </template>
            </tooltip>
            <text-field
              v-model="getCurrentUserEmail"
              input-type="text"
              height="40"
              :disabled="true"
              :required="true"
              class="disabledColor"
            />
          </div>
          <div class="pt-2 text-body-1 mb-2">
            What categories do you wish to receive?
          </div>
          <div>
            <v-treeview
              v-if="alertsSectionGroup.length != 0"
              v-model="tree"
              :items="alertsSectionGroup"
              item-key="show"
              open-all
              disabled
              class="header-class"
            >
              <template v-slot:append="{ item }">
                <hux-switch
                  v-if="item.name != 'categories'"
                  v-model="item.show"
                  false-color="var(--v-black-lighten4)"
                  :width="item.show ? '80px' : '100px'"
                  :switch-labels="switchLabel"
                  :class="item.show ? 'w-75' : 'w-97'"
                  @change="toggleIndividualSwitch($event, item)"
                />
              </template>
              <template v-slot:label="{ item }">
                <span
                  :class="
                    item.name != 'categories'
                      ? 'text-body-1'
                      : 'text-body-2 black--text text--lighten-4'
                  "
                  :data-type="item.type"
                  >{{ item.label | TitleCase }}</span
                >
              </template>
            </v-treeview>
          </div>
        </div>
        <div v-else>
          <v-alert outlined tile class="yellow lighten-1 black--text h-60">
            <div class="d-flex align-center">
              <div class="mr-3 mt-n2">
                <icon type="bulb" :size="36" color="yellow" />
              </div>
              <p class="text-body-1 ma-0 mt-n3">
                You have opted out from receiving alerts &amp; notifications.
              </p>
            </div>
          </v-alert>
        </div>
      </div>
    </template>
    <template #footer-left>
      <hux-button
        variant="white"
        size="large"
        :is-tile="true"
        class="mr-2 btn-border box-shadow-none"
        @click="closeDrawer"
      >
        <span class="primary--text">Cancel</span>
      </hux-button>
    </template>
    <template #footer-right>
      <div class="d-flex align-baseline">
        <hux-button
          variant="primary"
          size="large"
          :is-tile="true"
          data-e2e="save"
          @click="saveChanges"
        >
          Save
        </hux-button>
      </div>
    </template>
  </drawer>
</template>

<script>
import Drawer from "@/components/common/Drawer"
import Icon from "@/components/common/Icon"
import HuxSwitch from "@/components/common/Switch.vue"
import TextField from "@/components/common/TextField.vue"
import { mapActions, mapGetters } from "vuex"
import { formatText, formatInnerHTML } from "@/utils"
import Tooltip from "@/components/common/Tooltip.vue"
import HuxButton from "@/components/common/huxButton.vue"

export default {
  name: "AlertConfigureDrawer",
  components: {
    Drawer,
    Icon,
    HuxSwitch,
    TextField,
    Tooltip,
    HuxButton,
  },

  props: {
    value: {
      type: Boolean,
      required: true,
      default: false,
    },
  },
  data() {
    return {
      localDrawer: this.value,
      currentAlertConf: {},
      showAlerts: true,
      tree: null,
      switchLabelFullAlerts: [
        {
          condition: true,
          label: "YES",
        },
        {
          condition: false,
          label: "NO",
        },
      ],
      switchLabel: [
        {
          condition: true,
          label: "OPT IN",
        },
        {
          condition: false,
          label: "OPT OUT",
        },
      ],
      updatedConfiguration: {},
      alertsSectionGroup: [],
    }
  },

  computed: {
    ...mapGetters({
      getCurrentUserEmail: "users/getEmailAddress",
      getAlerts: "users/getUserAlerts",
    }),
  },

  watch: {
    value: function () {
      this.localDrawer = this.value
    },
    localDrawer: function () {
      this.$emit("input", this.localDrawer)
    },
    users: function () {
      this.updateUsers()
    },
  },
  mounted() {
    this.setDefaultConfig()
  },
  updated() {
    if (this.localDrawer) {
      this.mapAlertSectionGroups()
      this.maintainTreeStyles()
    }
  },
  methods: {
    ...mapActions({
      updateUserPreferences: "users/updateUserPreferences",
      getUsersNoti: "notifications/getAllUsers",
    }),
    formatInnerHTML: formatInnerHTML,
    async updateUsers() {
      await this.getUsersNoti()
    },
    closeDrawer() {
      this.localDrawer = false
    },
    saveChanges() {
      this.localDrawer = false
      this.formatFinalResponse()
      this.updateUserPreferences(this.updatedConfiguration)
      this.$emit("onDrawerClose")
    },
    maintainTreeStyles() {
      this.$nextTick(function () {
        document
          .getElementsByClassName("v-treeview-node")
          .forEach((element, index) => {
            if (element.childNodes.length == 2) {
              let elem = element.childNodes[0]
              if (index == 0) {
                elem.style["border-top"] = "none"
                elem.style["border-top-left-radius"] = "12px"
                elem.style["border-top-right-radius"] = "12px"
                elem.style["min-height"] = "30px"
                elem.style["height"] = "30px"
              }
              elem.style["background"] = "#F9FAFB"
            }
          })
      })
    },
    mapAlertSectionGroups() {
      this.currentAlertConf = this.getAlerts
      if (this.checkIfconfigExited(this.currentAlertConf)) {
        this.setAlertConfiguration()
      } else {
        this.showAlerts = false
      }
    },
    // Recursion for setting nested output structure
    recursiveBinding(data, alerts) {
      let parentData = data.children
      for (let i = 0; i < parentData.length; i++) {
        alerts[parentData[i].name] = {}
        if (parentData[i].show && !parentData[i].isDeepChild) {
          this.recursiveBinding(parentData[i], alerts[parentData[i].name])
        } else if (!parentData[i].show && !parentData[i].isDeepChild) {
          alerts[parentData[i].name] = {}
        } else {
          alerts[parentData[i].name] = parentData[i].show
        }
      }
      return alerts
    },

    checkIfconfigExited(entity) {
      return entity ? Object.keys(entity).length !== 0 : false
    },

    toggleIndividualSwitch(event, item) {
      this.manualToggleChanges(event, item)
      this.formatFinalResponse()
      this.maintainTreeStyles()
    },

    formatFinalResponse() {
      this.updatedConfiguration = {}
      this.updatedConfiguration.alerts = this.showAlerts
        ? this.recursiveBinding(this.alertsSectionGroup[0], {})
        : {}
    },

    manualToggleChanges(flag, item) {
      if (item && !item.isDeepChild) {
        item.children.forEach((data) => {
          if (data.children.length > 0) {
            // Recursion for setting event flag to deepest level
            data.show = flag
            this.manualToggleChanges(flag, data)
          } else {
            data.show = flag
          }
        })
      }
    },

    toggleMainSwitch(value) {
      if (value) {
        this.setDefaultConfig()
      } else {
        this.updatedConfiguration = {}
      }
      this.maintainTreeStyles()
    },

    // Recursion for getting nested input structure
    setAllKeyMapping(basicEntity, parentObj) {
      for (const [key, value] of Object.entries(basicEntity)) {
        let childObj = {
          label: formatText(key),
          name: key,
          isDeepChild: false,
          children: [],
        }
        if (typeof value === "object") {
          childObj.show = this.checkOptStatus(value)
          childObj.isDeepChild = false
          childObj.children = []
          parentObj.children.push(childObj)
          this.setAllKeyMapping(value, childObj)
        } else {
          childObj.show = value
          childObj.isDeepChild = true
          parentObj.children.push(childObj)
        }
      }
      return parentObj
    },
    setAlertConfiguration() {
      let parentObj = {
        label: "Categories",
        name: "categories",
        show: true,
        children: [],
      }
      this.setAllKeyMapping(this.currentAlertConf, parentObj)
      this.alertsSectionGroup = [parentObj]
    },
    checkOptStatus(value) {
      let convertedValue = JSON.stringify(value)
      return convertedValue.indexOf("true") != -1 ? true : false
    },
    setDefaultConfig() {
      this.alertsSectionGroup = [
        {
          label: "Categories",
          name: "categories",
          show: true,
          children: [
            {
              label: "Data management",
              name: "data_management",
              parent: null,
              show: true,
              children: [
                {
                  label: "Data sources",
                  name: "data_sources",
                  parent: "data_management",
                  show: true,
                  isDeepChild: false,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "data_sources",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "data_sources",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "data_sources",
                      show: true,
                      isDeepChild: true,
                    },
                  ],
                },
              ],
            },
            {
              label: "Decisioning",
              name: "decisioning",
              parent: null,
              show: true,
              children: [
                {
                  label: "Models",
                  name: "models",
                  parent: "decisioning",
                  show: true,
                  isDeepChild: false,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "models",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "models",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "models",
                      show: true,
                      isDeepChild: true,
                    },
                  ],
                },
              ],
            },
            {
              label: "Orchestration",
              name: "orchestration",
              parent: null,
              show: true,
              children: [
                {
                  label: "Destinations",
                  name: "destinations",
                  parent: "orchestration",
                  isDeepChild: false,
                  show: true,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "destinations",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "destinations",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "destinations",
                      show: true,
                      isDeepChild: true,
                    },
                  ],
                },
                {
                  label: "Delivery",
                  name: "delivery",
                  parent: "orchestration",
                  show: true,
                  isDeepChild: false,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "delivery",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "delivery",
                      isDeepChild: true,
                      show: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "delivery",
                      isDeepChild: true,
                      show: true,
                    },
                  ],
                },
                {
                  label: "Audiences",
                  name: "audiences",
                  parent: "orchestration",
                  show: true,
                  isDeepChild: false,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "audiences",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "audiences",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "audiences",
                      show: true,
                      isDeepChild: true,
                    },
                  ],
                },
                {
                  label: "Engagements",
                  name: "engagements",
                  parent: "orchestration",
                  show: true,
                  isDeepChild: false,
                  children: [
                    {
                      label: "Critical",
                      name: "critical",
                      parent: "engagements",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Success",
                      name: "success",
                      parent: "engagements",
                      show: true,
                      isDeepChild: true,
                    },
                    {
                      label: "Informational",
                      name: "informational",
                      parent: "engagements",
                      show: true,
                      isDeepChild: true,
                    },
                  ],
                },
              ],
            },
          ],
        },
      ]
    },
  },
}
</script>
<style lang="scss" scoped>
.black-border {
  border-radius: 50%;
  border: 0.5px solid var(--v-black-base);
}
.full-alert {
  justify-content: space-between;
  align-items: center;
  height: 40px;
}
.w-53 {
  width: 53px;
}
.w-75 {
  width: 75px;
}
.w-97 {
  width: 97px;
}
.disabledColor .v-input__slot {
  background-color: var(--v-primary-lighten1) !important;
}
.h-77 {
  height: 77px;
}
.h-60 {
  height: 60px;
}
::v-deep .v-treeview .v-treeview-node__root {
  border-top: 1px solid var(--v-black-lighten2);
  height: 45px;
  padding: 0;
}
::v-deep .v-treeview-node__toggle {
  display: none;
}
::v-deep .v-treeview-node__content {
  padding: 10px 24px 11px 16px;
  height: 45px;
  margin-left: 0 !important;
}
.header-class {
  border: 1px solid var(--v-black-lighten2);
  border-radius: 12px;
}
::v-deep .v-text-field__slot {
  input {
    color: var(--v-black-lighten4) !important;
  }
}
::v-deep .v-treeview-node__content {
  .v-treeview-node__label {
    span {
      color: var(--v-black-base) !important;
    }
  }
  .v-treeview-node__append {
    .v-input {
      .v-input__control {
        .v-input__slot {
          .v-input--selection-controls__input {
            .v-input--switch__thumb {
              &:last-child {
                right: 1px !important;
                &.primary--text {
                  right: 0px !important;
                }
              }
            }
          }
        }
      }
    }
  }
}
::v-deep .v-treeview {
  & > .v-treeview-node {
    & > .v-treeview-node__root {
      & > .v-treeview-node__content {
        & > .v-treeview-node__label {
          & > span {
            color: var(--v-black-lighten4) !important;
          }
        }
      }
    }
  }
}
</style>
