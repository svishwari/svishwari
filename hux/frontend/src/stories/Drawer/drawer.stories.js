import Drawer from "./Drawer2.vue"
import HuxButton from "../huxButton/huxButton2.vue"
import { action } from "@storybook/addon-actions"
import AllIcons from "../icons/Icons"
import Page2 from "../Page/Page2.vue"

export default {
  component: Drawer,
  title: "NewComponents/Drawer",
  argTypes: {
    width: {
      control: { type: "number" },
    },
    loading: { table: { disable: true } },
    title: { control: { type: "text" } },
    drawerContent: { control: { type: "text" } },
    iconType: {
      defaultValue: "mapping",
      options: AllIcons,
      control: {
        type: "select",
      },
    },
    status: {
      control: { type: "select" },
      options: ["Text", "Empty", "Error"],
    },
    secondaryButtonText: { control: { type: "text" } },
    secondPrimaryButtonText: { control: { type: "text" } },
    primaryButtonText: { control: { type: "text" } },
    footerTextField: { control: { type: "text" } },
  },
  args: {
    loading: false,
    width: 400,
    title: "Title",
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/4qNDv9mcu1ZWZkZxO3fVpP/6.0-Release?node-id=11331%3A256832",
    },
  },
}

const Template = (args, { argTypes }) => ({
  components: { Drawer, HuxButton, Page2 },
  props: Object.keys(argTypes),
  methods: {
    onCancel: action("onCancel"),
    onCreate: action("onCreate"),
    toggleDrawer: action("toggleDrawer"),
    onClose: action("onClose"),
  },
  argTypes: {
    onClose: {},
  },
  data() {
    return {
      toggleDrawer: false,
    }
  },
  template: `
    <div>
    <hux-button @click="toggleDrawer = !toggleDrawer"> Toggle Drawer</hux-button>
    <drawer
      v-model="toggleDrawer"
      v-bind="$props"
      v-on="$props"
      @onClose="onClose()"
    >
      <template #drawerContent>
        <div class="new-b1 pa-6" v-if="$props.status == 'Text'">
          {{$props.drawerContent}}
        </div>
        <div class="new-b1 pa-6" v-if="$props.status == 'Error'" >
          <page2 errorState="true" />
        </div>
        <div class="new-b1 pa-6" v-if="$props.status == 'Empty'" >
          <page2 emptyState="true" />
        </div>
      </template>
    </drawer>
    </div>`,
})

export const drawer = Template.bind({})
