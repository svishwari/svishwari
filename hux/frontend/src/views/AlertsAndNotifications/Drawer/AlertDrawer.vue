<template>
  <drawer
    v-model="localDrawer"
    :content-padding="'pa-0'"
    :content-header-padding="'px-3'"
  >
    <template #header-left>
      <div
        v-if="
          notificationData &&
          notificationData.notification_type &&
          notificationData.id
        "
        class="d-flex align-center"
      >
        <icon
          :type="
            notificationData.notification_type === 'Success'
              ? 'success_new'
              : notificationData.notification_type
          "
          :size="32"
          :color="getIconColor(notificationData.notification_type)"
          :variant="getIconVariant(notificationData.notification_type)"
          class="d-block mr-2"
        />
        <h3 class="text-h2 ml-1 black--text text--darken-4">
          Alert ID: {{ notificationData.id }}
        </h3>
      </div>
    </template>
    <template #default>
      <div
        v-if="notificationData && notificationContent"
        class="pl-7 pt-1 pr-5"
      >
        <div v-for="data in notificationContent" :key="data.id" class="mt-5">
          <div class="text-body-1 black--text">
            {{ data.title }}
          </div>
          <div class="text-body-1 black--text text--lighten-4 mt-1">
            {{ data.value | Empty("-") }}
          </div>
        </div>
      </div>
    </template>
  </drawer>
</template>

<script>
import { mapGetters } from "vuex"
import Drawer from "@/components/common/Drawer"
import Icon from "@/components/common/Icon"

export default {
  name: "AlertDrawer",
  components: {
    Drawer,
    Icon,
  },

  props: {
    value: {
      type: Boolean,
      required: true,
      default: false,
    },
    notificationId: {
      type: String,
      required: false,
      default: "",
    },
  },
  data() {
    return {
      localDrawer: this.value,
    }
  },

  computed: {
    ...mapGetters({
      getSingleNotification: "notifications/single",
    }),

    notificationData() {
      return this.getSingleNotification(this.notificationId)
    },

    notificationContent() {
      if (this.notificationData) {
        return [
          {
            id: 1,
            title: "Full Description",
            value: this.notificationData.description,
            subLabel: null,
          },
          {
            id: 2,
            title: "Category",
            value: this.notificationData.category,
          },
          {
            id: 3,
            title: "Date",
            value: this.formattedDate(this.notificationData.created),
          },
          {
            id: 4,
            title: "User",
            value: this.notificationData.username,
            subLabel: null,
          },
        ]
      }
      return []
    },
  },

  watch: {
    value: function () {
      this.localDrawer = this.value
    },
    localDrawer: function () {
      this.$emit("input", this.localDrawer)
    },
  },

  methods: {
    formattedDate(value) {
      if (value) {
        return this.$options.filters.Date(value, "MMM DD, YYYY h:mm A")
      }
      return "-"
    },
    getIconColor(value) {
      if (value) {
        return value === "Success"
          ? "success"
          : value === "Critical"
          ? "error"
          : "primary"
      }
    },

    getIconVariant(value) {
      if (value) {
        return value === "Informational" ? "lighten6" : "base"
      }
    },
  },
}
</script>
