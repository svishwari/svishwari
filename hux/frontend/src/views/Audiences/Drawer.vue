<template>
  <div>
    <Drawer v-model="toggleDrawer">
      <template #header-left>
        <h3 class="text-h3">Add audiences to this engagement</h3>
      </template>

      <template #default>
        <v-progress-linear :active="loading" :indeterminate="loading" />

        <div class="pa-8">
          <!-- TODO: HUS-229, HUS-445 -->
          <v-btn disabled tile color="primary" class="mb-4">
            <v-icon>mdi-plus</v-icon>
            New audience
          </v-btn>

          <DataCards
            :items="audiences"
            :fields="[
              {
                key: 'name',
                label: 'Name',
                sortable: true,
              },
              {
                key: 'size',
                label: 'Target size',
                sortable: true,
              },
              {
                key: 'manage',
                sortable: false,
              },
            ]"
          >
            <template #field:size="row">
              <v-menu offset-y open-on-hover>
                <template #activator="{ on }">
                  <span v-on="on">
                    {{ row.value | Numeric(true, true) }}
                  </span>
                </template>
                <div class="px-4 py-3 text-caption white">
                  {{ row.value | Numeric }}
                </div>
              </v-menu>
            </template>

            <template #field:manage="row">
              <div class="d-flex align-center justify-end">
                <v-btn
                  v-if="isAdded(row.item)"
                  color="secondary"
                  width="100"
                  @click="remove(row.item)"
                >
                  <v-icon small class="mr-1">mdi-check</v-icon>
                  Added
                </v-btn>
                <v-btn
                  v-else
                  outlined
                  color="lightGrey"
                  width="100"
                  @click="add(row.item)"
                >
                  <span class="darkGrey--text">Add</span>
                </v-btn>
              </div>
            </template>
          </DataCards>
        </div>
      </template>

      <template #footer-left> {{ audiences.length }} results </template>
    </Drawer>
  </div>
</template>

<script>
import { mapGetters, mapActions } from "vuex"
import DataCards from "@/components/common/DataCards.vue"
import Drawer from "@/components/common/Drawer.vue"

export default {
  name: "AudiencesDrawer",

  components: {
    DataCards,
    Drawer,
  },

  props: {
    value: {
      type: Object,
      required: true,
    },

    toggle: {
      type: Boolean,
      required: false,
      default: false,
    },
  },

  data() {
    return {
      toggleDrawer: false,
      loading: false,
    }
  },

  watch: {
    toggle(value) {
      this.toggleDrawer = value
    },
  },

  computed: {
    ...mapGetters({
      audiences: "audiences/list",
    }),
  },

  methods: {
    ...mapActions({
      getAudiences: "audiences/getAll",
    }),

    isAdded(audience) {
      return Boolean(this.value[audience.id])
    },

    add(audience) {
      this.$set(this.value, audience.id, {
        id: audience.id,
        name: audience.name,
        size: audience.size,
      })
    },

    remove(audience) {
      this.$delete(this.value, audience.id)
    },
  },

  async mounted() {
    this.loading = true
    await this.getAudiences()
    this.loading = false
  },
}
</script>
