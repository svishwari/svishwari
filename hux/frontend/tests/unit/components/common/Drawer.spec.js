import Drawer from "@/components/common/Drawer"
import { mount, shallowMount } from "@vue/test-utils"
import Vuetify from 'vuetify';

const vuetify = new Vuetify();

describe("Drawer", () => {

  test("Drawer calls function on close", async () => {

    const wrapper = mount(Drawer,{
        propsData: {
            value: true,
        },
        vuetify,
    })
    await wrapper.setData({ localDrawer: false })
    expect(  typeof wrapper.emitted().onClose === 'function')
    expect(wrapper).toMatchSnapshot()
  })
})
