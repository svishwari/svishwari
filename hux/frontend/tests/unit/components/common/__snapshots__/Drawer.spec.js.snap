// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Drawer calls function on close 1`] = `
<v-navigation-drawer-stub app="true" mobilebreakpoint="1264" hideoverlay="true" floating="true" height="100vh" minivariantwidth="56" right="true" src="" tag="nav" temporary="true" width="600" style="transition-duration: 0.5s;">
  <v-toolbar-stub height="70" width="100%" tag="header" extensionheight="48" src="" class="drawer-header no-shadow border-bottom">
    <v-toolbar-title-stub tag="div"> </v-toolbar-title-stub>
    <!---->
  </v-toolbar-stub>
  <v-progress-linear-stub color="primary" value="0" buffervalue="100" height="4"></v-progress-linear-stub>
  <div class="drawer-content contentPadding drawer-content-without-footer"></div>
  <!---->
</v-navigation-drawer-stub>
`;
